### ISSUE TITLE
Support BEFORE hooks

### ISSUE THREAD

**Issue by maps82** [tj-django/django-model-subscription#105]
Hello!

I noticed that only the AFTER_* hooks of django-lifecycle are implemented.

Wouldn't it be useful to also support the BEFORE_* hooks?

E.g. I have an attachment file model. If an object get's deleted I'd like to use a BEFORE_DELETE hook to also delete the related file in the filesystem. With OperationType.DELETE this is not possible as the object is removed and I can't access it's fields in the hook anymore.

**Comment by jackton1** [comment]
Yeah correct, plan to support before an after decorators but for now only the pk would be set to None

Using 
```
@before_delete_subscription
def (..):
    ...


@after_delete_subscription
def (..):
    ...
```
